src/main/java/org/cyclops/integrateddynamics/block/BlockDryingBasin.java

```diff
@@ -60,20 +60,25 @@ public boolean onBlockActivated(World world, BlockPos blockPos, IBlockState bloc
         if(world.isRemote) {
             return true;
         } else {
-            ItemStack itemStack = player.inventory.getCurrentItem();
             TileDryingBasin tile = TileHelpers.getSafeTile(world, blockPos, TileDryingBasin.class);
             if (tile != null) {
+                ItemStack itemStack = player.inventory.getCurrentItem();
                 IFluidHandler itemFluidHandler = FluidUtil.getFluidHandler(itemStack);
                 SingleUseTank tank = tile.getTank();
                 ItemStack tileStack = tile.getStackInSlot(0);
-                if ((itemStack == null || (ItemStack.areItemsEqual(itemStack, tileStack) && ItemStack.areItemStackTagsEqual(itemStack, tileStack) && itemStack.stackSize < itemStack.getMaxStackSize())) && tileStack != null) {
+
+                if ((itemStack == null || (ItemStack.areItemStacksEqual(itemStack, tileStack) && tileStack.stackSize < tileStack.getMaxStackSize())) && tileStack != null) {
```

Comment:
This case never occurs, because of `ItemStack.areItemStacksEqual` also checking stacksizes.
It appears however that the second case you added with `addItemStackToInventory` is enough, and the vanilla code correctly handles this as intended.

So you can remove this first if-block completely. You can double check to make sure I didn't forget any edge-case.
That if-block can then also be removed for the Squeezer.